### DISCUSSION TITLE
Local or Cloud Dependencies

### DISCUSSION THREAD

**Discussion by teek541** [NSPManager/NSPanelManager#193] [Q&A] [answered]
Hey guys, I've been looking at getting a few NS Panels for some specific use cases.  I always thought that NS Panels required cloud access. Does this allow for 100% local use with Home Assistant through MQTT?  It looks more straight forward than the other solution of custom flashing the NS Panels, side loading a launcher, and installing the HA Companion app.

**Comment by aaronouthier** [comment]
I believe I can answer that. Please note that I am not one of the Devs, so defer to them for definitive info.

Yes, data stays in the local network with this setup. It requires running their server-side software, either as a Home Assistant add-on or as a stand-alone container in Docker, etc. I run mine on my Proxmox home lab server.

You can't just flash the firmware and have it work with all features. The relays will work without the server-side of things, but without that server software, little-else will.

**Reply by teek541** [reply]
Thanks for the reply.  Yeah, I figured this add-on was required on Home Assistant along with MQTT and the Flashing of the firmware.  I'm just debating if I want to go this route or go full on Home Assistant Companion app.  This seems a bit simpler and will probably be supported a lot easier in the long run.  I'm only using NS panels as advanced light switch/controls for a few things.  Primarily controlling lights in my back yard through other integrations (Smart Switches, Pool Integration, Generic Thermostat for Whole House Fan, etc.)  I think I'll probably keep the default NS Panel for the time being and use this integration.  If the need for a completely custom dashboard arises in the future I'll probably go full on HA Panel with it.

Thanks for confirming this doesn't require a cloud connection, just a local docker/mqtt connection.

**Reply by aaronouthier** [reply]
I believe you misunderstood something. There is no Home Assistant app or anything else that can be side-loaded onto the Regular NSPanel. There are 2 different models: The NSPanel and the NSPanel pro. The Pro model runs Android, and because of this, you can side-load apps. The regular NSPanel (with the 2 buttons) does not, and can not.

Unless I misunderstood what you are saying?

**Reply by teek541** [reply]
I may be misunderstanding.  I ordered the Pro models, I haven't set them up yet.  Does the NS Panel Manager (loaded as a HA Plugin) work for the pro models w/o the need for cloud integrations?

In the long run, if I need a HA Dashboard for something (thermostat controls, alarm panel, etc.) I was planning on side loading the HA Companion App on my Pro model.  Otherwise I was going to try this NS Panel Manager as a Plugin for HA.

**Comment by aaronouthier** [comment]
Pro model is not compatible with this project, to the best of my knowledge. Only option is the HA App for Android, and the Pro model is always cloud based, until someone takes the time to create custom firmware for it. The Regular and Pro models have different CPUs and hardware, so adapting firmware from one to work on the other is a big chore.

**Reply by teek541** [reply]
My mistake, thanks for clarifying.  Looks like I'll be making more dashboards and side loading in my future for the Pro.  By doing custom launchers and such I believe you can make the Pro model local by running the HA companion app.

**Reply by aaronouthier** [reply]
The companion app would be local, but the base Android OS and the customizations made by Sonoff, as well as any other Apps installed on the Pro would be either cloud or local, according to their individual manufacturer's programming.

**Comment by aaronouthier** [comment]
As the actual Developers of this project are in the Netherlands, I believe, they may be asleep. Might be good to wait for them to clarify or validate what I've said. While my info is correct to the best of my knowledge, as I said, I'm not the Dev, and am only human...

**Comment by tpanajott** [comment, marked as answer]
Thanks @aaronouthier for all your help with answering questions. Though we are based in Sweden, same timezone though :)

Regarding your question. This firmware is for regular NSPanels, the ones using an ESP32 chip inside and not running Android. For the android based devices I've not had much look at them but I believe @aaronouthier answers are sufficient. We've though about building our own web page that would behave the same way so that it can be run in kiosk mode and have the same look as regular NSPanels though this is just an idea for now.

The only thing this project requires are
1. Home Assistant or OpenHAB.
2. An MQTT server.
3. Somewhere to run the manager container (can be done as an HA addon or standalone container).

**Reply by aaronouthier** [reply]
You forgot one more in your list of requirements: A "non-pro" NSPanel 😉.

**Reply by teek541** [reply]
Yes, for us dumb new NSPanel users who didn't know the difference between Pro and Non-Pro.

I would say that offering a webpage kiosk mode would be helpful, then you could just point it to whatever environment you want.  HA or a NSPanel server.

**Reply by aaronouthier** [reply]
Uninformed is not the same as dumb. Sometimes we might feel "dumb" for not having all of the right info, but that is temporary.
